protocol-impl/src/main/java/io/zeebe/protocol/impl/record/value/job/JobBatchRecord.java

```diff
@@ -159,7 +159,20 @@ public int getMaxJobsToActivate() {
 
   @Override
   public List<JobRecordValue> getJobs() {
-    return StreamSupport.stream(jobsProp.spliterator(), false).collect(Collectors.toList());
+
+    return StreamSupport.stream(jobsProp.spliterator(), false)
+        .map(
+            r ->
+                new JobRecord()
+                    .setRetries(r.getRetries())
+                    .setDeadline(r.getDeadlineLong())
+                    .setErrorMessage(r.getErrorMessageBuffer())
+                    .setWorker(r.getWorkerBuffer())
+                    .setVariables(r.getVariablesBuffer())
```

Comment:
We have still here the problem that the buffers are wrapped, which means they use internally the same reference.

Reply:
We have to create a new byte array and write the values into this array. After that we can wrap this array. Here is the code how I did it on the DeploymentRecord for the worfklows

```
  @Override
  public List<DeployedWorkflow> getDeployedWorkflows() {
    final List<DeployedWorkflow> workflows = new ArrayList<>();

    final Iterator<Workflow> iterator = workflowsProp.iterator();
    while (iterator.hasNext()) {
      final Workflow workflow = iterator.next();

      final byte[] bytes = new byte[workflow.getLength()];
      final UnsafeBuffer copyBuffer = new UnsafeBuffer(bytes);
      workflow.write(copyBuffer, 0);

      final Workflow copiedWorkflow = new Workflow();
      copiedWorkflow.wrap(copyBuffer);
      workflows.add(copiedWorkflow);
    }

    return workflows;
  }
```